### ISSUE TITLE
Impossible to make a string containing \"

### ISSUE THREAD

**Issue by markasoftware** [AmethystGear/JACL#2]
`hello \"` => `hello "`
`hello \\"` => error
`hello \\\"` => `hello \\"`

Simplest solution seems to be changing the regex to `^(["'])(\\\"|.)*?\1`, which means `hello \\"` turns into `hello \"` (the `|` part of the regex always prefers the left side, ie short circuit). As an added bonus the unnecessary lookahead is gone. (though this doesn't solve issue #3)

**Comment by AmethystGear** [comment]
Oof.
Looks like my regex skills need improvement. Nice work.

**Comment by AmethystGear** [comment]
Ok, this is weird.
Your regex doesn't work when I put it in my python code, but it works on https://regex101.com/.
After trying to understand why, I gave up and searched this on stackoverflow.
i'm now using 
```"(?:[^"\\]*(?:\\.)?)*"```
Got it from https://stackoverflow.com/questions/249791/regex-for-quoted-string-with-escaping-quotes, Marc-André Poulin's answer.
It seems to work for both this issue and #3. I added a test case to `test.jacl` for it. I'll probably get back to understanding how exactly this works eventually, but it will suffice for now.
